### PR TITLE
Habib/pip calculator

### PR DESCRIPTION
**Pull request by ShuvoHabib**
Changes:

Added PIP calculator in the list of existing calculators

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/trader-tools/common/_utility.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    return errors
+}
+
+const getMarginCurrency = (symbol) => {
+    let currency = 'USD'
+    if (symbol.market === 'synthetic_indices' || symbol.market === 'commodities') {
+        currency = 'USD'
+    }
```
Comment (yashim-deriv) [inline src/pages/trader-tools/common/_utility.js:50]:
```suggestion
```
Should this be omitted since the default is already 'USD' ?

Comment (ShuvoHabib) [reply src/pages/trader-tools/common/_utility.js]: Cool.. Resolved

**Comment by sean-binary** [comment]
@ShuvoHabib This section should in fact be in text strings instead of an image, so that we can translate it.
image

File: src/pages/trader-tools/pip-calculator/_pip-calculator.js [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+                                    <QueryImage
+                                        data={data.pip_forex_info_mobile}
+                                        alt={localize('Pip Forex Info')}
+                                    />
+                                </Show.Mobile>
+
+                                <Text mb="1.6rem" mt="1.6rem">
+                                    <Localize
+                                        translate_text="So your pip value is 2 USD."
+                                        components={[<strong key={0} />]}
```
Comment (sean-binary) [inline src/pages/trader-tools/pip-calculator/_pip-calculator.js:411]: Same here.

File: src/pages/trader-tools/pip-calculator/_pip-calculator.js [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+                                    <QueryImage
+                                        data={data.pip_info_mobile}
+                                        alt={localize('Pip Info')}
+                                    />
+                                </Show.Mobile>
+
+                                <Text mb="1.6rem" mt="1.6rem">
+                                    <Localize
+                                        translate_text="So your pip value is 0.01 USD."
+                                        components={[<strong key={0} />]}
```
Comment (sean-binary) [inline src/pages/trader-tools/pip-calculator/_pip-calculator.js:369]: I don't see the `strong` tag being used 👀  We can replace this with a regular `localize` instead.

File: src/pages/trader-tools/pip-calculator/_pip-calculator.js [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+                        </Text>
+                        <Text mb="1.6rem">
+                            <Localize
+                                translate_text="<0>Pip value = point value × volume × contract size</0>"
+                                components={[<strong key={0} />]}
+                            />
+                        </Text>
+
+                        <Text mb="1.6rem">
+                            <Localize translate_text="For synthetic accounts, the pip value is calculated in USD." />
```
Comment (sean-binary) [inline src/pages/trader-tools/pip-calculator/_pip-calculator.js:319]: Let's just use the regular `localize` here.

File: src/pages/trader-tools/pip-calculator/_pip-calculator.js [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+                        </Formik>
+                    </FormWrapper>
+
+                    <RightContent>
+                        <Header as="h3" type="section-title" mb="0.8rem">
+                            {localize('How to calculate pip value')}
+                        </Header>
+
+                        <Text>
+                            <Localize translate_text="The pip value for a contract on DMT5 is calculated based on this formula:" />
```
Comment (sean-binary) [inline src/pages/trader-tools/pip-calculator/_pip-calculator.js:309]: Let's just use the regular `localize` here.

**Comment by ShuvoHabib** [comment]
> @ShuvoHabib This section should in fact be in text strings instead of an image, so that we can translate it. This applies for both synthetic & financial (desktop & mobile).
> 
> image

Hi @sean-binary  Resolved

File: src/pages/trader-tools/common/_utility.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
+import validation from './_validation';
+
+const getMargin = (values) => {
```
Comment (sean-binary) [inline src/pages/trader-tools/common/_utility.js:3]: Some of these utility functions seem to be pip calculator specific and some seem to be shared with margin calculator. Could you help to keep specific ones in `_pip-calculator` and use the reusable ones for `_margin-calculator` as well? 🙏

File: src/pages/trader-tools/pip-calculator/_pip-calculator.js [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+                                            <span>
+                                                {localize('One standard lot of forex = 100,000 units')}
+                                            </span>
+                                        </li>
+                                    </StyledOl>
+                                </FormulaText>
+
+                                <Text mb="1.6rem" mt="1.6rem">
+                                    <Localize
+                                        translate_text="So your pip value is <0>2 USD.</0>"
```
Comment (sean-binary) [inline src/pages/trader-tools/pip-calculator/_pip-calculator.js:420]:
```suggestion
                                        translate_text="So your pip value is <0>2 USD</0>."
```

File: src/pages/trader-tools/common/_utility.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+    if (symbol_error) {
+        errors.symbol = symbol_error
+    }
+    if (volume_error) {
+        errors.volume = volume_error
+    }
+    if (pointValue_error) {
+        errors.pointValue = pointValue_error
+    }
+
```
Comment (yashim-deriv) [inline src/pages/trader-tools/common/_utility.js:164]:
```suggestion
```

File: src/pages/trader-tools/common/_utility.js [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+    if (symbol_error) {
+        errors.symbol = symbol_error
+    }
+    if (volume_error) {
+        errors.volume = volume_error
+    }
+    if (assetPrice_error) {
+        errors.assetPrice = assetPrice_error
+    }
+
```
Comment (yashim-deriv) [inline src/pages/trader-tools/common/_utility.js:140]:
```suggestion
```

File: src/pages/trader-tools/common/_utility.js (hunk 8 above)
Comment (yashim-deriv) [inline src/pages/trader-tools/common/_utility.js:164]:
```suggestion
```

File: src/pages/trader-tools/pip-calculator/_pip-calculator.js [hunk 10]
```diff
@@ -0,0 +1,446 @@
...
+                                        <li>
+                                            <span>
+                                                {localize('The contract size is one standard lot of Volatility 75 index = 1')}
+                                            </span>
+                                        </li>
+                                    </StyledOl>
+                                </FormulaText>
+                                <Text mb="1.6rem" mt="1.6rem">
+                                    <Localize
+                                        translate_text="So your pip value is <0>0.01 USD.</0>"
```
Comment (sean-binary) [inline src/pages/trader-tools/pip-calculator/_pip-calculator.js:371]:
```suggestion
                                        translate_text="So your pip value is <0>0.01 USD</0>."
```

File: src/pages/trader-tools/common/_utility.js [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+        formula = specialFormula
+    }
+    return formula
+}
+
+const getMargin = (values) => {
+    alert(values);
+    const { volume, assetPrice, leverage, contractSize } = values
+    const specialFormula = (volume * contractSize * assetPrice) / leverage.name
+    let margin_formula = rawCalculation(values, specialFormula)
```
Comment (yashim-deriv) [inline src/pages/trader-tools/common/_utility.js:30]:
```suggestion
    let margin_formula = rawCalculation(values, specialFormula)
    
```

**Review by yashim-deriv — COMMENTED** [review]
Just minor suggestions to white space usage (for consistency). Apart from that everything looks good. 🙏🏼

File: src/pages/trader-tools/common/_utility.js [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+
+const getSwapChargeForex = (values) => {
+    const { volume, pointValue, swapRate, contractSize } = values
+    const swap_formula_forex = volume * contractSize * pointValue * swapRate
+    return toFixed(swap_formula_forex)
+}
+
+const toFixed = (val) => {
+    return parseFloat(val.toFixed(3)).toLocaleString()
+}
```
Comment (sean-binary) [inline src/pages/trader-tools/common/_utility.js:58]: We can reuse this in `_margin-calculator` as well.

File: src/pages/trader-tools/common/_utility.js [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+    return errors
+}
+
+const numberWithCommas = (input) => {
+    return input.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
+}
+
+const numberSubmitFormat = (input) => {
+    return input.replace(/^0+(?!\.|$)/, '')
+}
```
Comment (sean-binary) [inline src/pages/trader-tools/common/_utility.js:178]: Same here. And I think a couple more.

**Review by sean-binary — APPROVED** [review]
Nice cleanup 👍

**Pull request merged**